scheduler/scheduler.cpp

```diff
@@ -1816,6 +1823,86 @@ static void trigger_exit(int signum)
     signal(signum, trigger_exit);
 }
 
+static void handle_scheduler_announce(char* buf, const char* netname, int schedbuflen, bool persistent_clients, struct sockaddr_in broad_addr)
+{
+    /* Another scheduler is announcing it's running, disconnect daemons if it has a better version
+       or the same version but was started earlier. */
+    if (!persistent_clients){
+        uint64_t tmp_time;
+        memcpy(&tmp_time, buf + 4, sizeof(uint64_t));
+        time_t other_time = tmp_time;
+        int other_scheduler_protocol = buf[3];
+        string recv_netname;
+        if (other_scheduler_protocol >= 36)
+        {
+            buf[schedbuflen] = '\0';
+            recv_netname = buf + 5 + sizeof(uint64_t);
+            if (recv_netname == netname)
+            {
+                if (other_scheduler_protocol > PROTOCOL_VERSION || (other_scheduler_protocol == PROTOCOL_VERSION && other_time < starttime))
+                {
+                    if (!css.empty() || !monitors.empty())
+                    {
+                        log_info() << "Scheduler from " << inet_ntoa(broad_addr.sin_addr)
+                            << ":" << ntohs(broad_addr.sin_port)
+                            << " (version " << int(other_scheduler_protocol) << ") has announced itself as a preferred"
+                            " scheduler, disconnecting all connections." << endl;
+                        while (!css.empty())
+                        {
+                            handle_end(css.front(), NULL);
+                        }
+                        while (!monitors.empty())
+                        {
+                            handle_end(monitors.front(), NULL);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                for(list<CompileServer *>::iterator it = css.begin();it != css.end();++it)
+                {
+                    if ((*it)->netname() == recv_netname)
```

Comment:
we had to add netname to login message so that we know it is OK to disconnect a client if their netname does not match with schedulers netname